src/main/java/org/terasology/HealthComponent.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.network.Replicate;
+import org.terasology.rendering.nui.properties.TextField;
+
+public class HealthComponent implements Component {
+
+    @Replicate
+    public int maxHealth = 20;
+    @Replicate
+    public float regenRate;
+    @Replicate
+    public float waitBeforeRegen;
+
+    @Replicate
+    public float fallingDamageSpeedThreshold = 20;
+    @Replicate
+    public float horizontalDamageSpeedThreshold = 20;
+    @Replicate
+    public float excessSpeedDamageMultiplier = 10f;
+
+    @Replicate
+    @TextField
+    public int currentHealth = 20;
+
+    // Regen info
+    public long nextRegenTick;
+
+    public boolean destroyEntityOnNoHealth;
+
+    public HealthComponent() {
```

Comment:
As a practice it may be wise to leave a comment on the no-arg constructor to highlight the reason it is here: if it isn't and there is an arg-constructor then the "invisible" no-arg constructor disappears. That then breaks stuff as some of the magic relies on there always being a no-arg constructor :-)